### PR TITLE
Specify endian-ness and compression algorithm in PROTOCOL.md

### PR DESCRIPTION
**Pull request by jwalton**
:)

### CODE REVIEW

File: PROTOCOL.md [hunk 1]
```diff
@@ -94,8 +96,8 @@ data frames the writer will send before blocking for acks.
 
 Payload:
 
-* 32bit unsigned payload length 
-* 'length' bytes of compressed payload
+* 32bit unsigned payload length
+* 'length' bytes of zlib compressed 'data' frames.
```
Comment (jordansissel) [inline PROTOCOL.md]: Compressed frames contain lumberjack frames; not required to be data frames.

Comment (jordansissel) [inline PROTOCOL.md]: However, given this is the current behavior, I'm OK with this as it is

**Comment by jordansissel** [comment]
@jwalton can you do step 2 of https://github.com/elasticsearch/logstash-forwarder/blob/master/CONTRIBUTING.md#contribution-steps?

**Comment by jwalton** [comment]
Done.

**Pull request closed without merging**